src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -1915,6 +1918,10 @@ impl Coordinator {
     ) -> Result<PeekStageTimestamp, AdapterError> {
         let source = self.view_optimizer.optimize(source)?;
 
+        let id_bundle = self
```

Comment:
This gets recomputed below in `peek_stage_finish`. I'm ok merging this as is, but if you feel ambitious this bundle could get stashed in the various PeekStage enum structs so we only compute it once.

Reply:
Good idea!

Reply:
It ends up being an option in `PeekStageFinish` because some other code paths got there seemingly without computing the id bundle. That means the _code_ is duplicated, but building the id bundle should only happen once.